cmake/resources/generate_resources.cmake

```diff
@@ -12,11 +12,30 @@ set(
 set(RES_IMAGE_EXCLUDE_FILTER ^linuxinstall/)
 
 file(GLOB_RECURSE RES_ALL_FILES RELATIVE "${RES_DIR}" LIST_DIRECTORIES false CONFIGURE_DEPENDS "${RES_DIR}/*")
-file(GLOB_RECURSE RES_IMAGE_FILES RELATIVE "${RES_DIR}" LIST_DIRECTORIES false CONFIGURE_DEPENDS "${RES_DIR}/*.png")
+file(GLOB_RECURSE RES_IMAGE_FILES RELATIVE "${RES_DIR}" LIST_DIRECTORIES false CONFIGURE_DEPENDS "${RES_DIR}/*.png" "${RES_DIR}/*.svg")
 
 list(REMOVE_ITEM RES_ALL_FILES ${RES_IGNORED_FILES})
 list(FILTER RES_IMAGE_FILES EXCLUDE REGEX ${RES_IMAGE_EXCLUDE_FILTER})
 
+# Remove files with the same name, regardless of extension
+foreach (_file ${RES_IMAGE_FILES})
+    get_filename_component(_ext "${_file}" EXT)
+    string(REPLACE "${_ext}" "" _var_name ${_file})
+
+    get_filename_component(_fileName "${_file}" NAME_WE)
+    if (${_fileName} STREQUAL "switch") # cuz it errors
+        list(REMOVE_ITEM RES_IMAGE_FILES ${_file})
+        continue()
+    endif()
+
+    if (${_var_name} IN_LIST DUPLICATE_CHECKER)
+        list(REMOVE_ITEM RES_IMAGE_FILES ${_file})
+        continue()
+    endif()
+
+    list(APPEND DUPLICATE_CHECKER ${_var_name})
+endforeach ()
```

Comment:
Why is this necessary?

Reply:
This was to get .svg files to work as well and to remove duplicate filenames or it would add both the .png and the .svg in the generate C++ files and error out cuz of duplicate